diff --git a/app/models/upload.rb b/app/models/upload.rb
index 273b969..02cf65c 100644
--- a/app/models/upload.rb
+++ b/app/models/upload.rb
@@ -32,8 +32,8 @@ def options
     {
       "Cache-Control" => "max-age=315360000, public",
       "Expires" => "Sun, 29 Jun 2036 17:48:34 GMT",
-      "x-amz-storage-class" => "REDUCED_REDUNDANCY"
+      "x-amz-storage-class" => ENV["AWS_S3_STORAGE_CLASS"] || "REDUCED_REDUNDANCY",
     }
   end
 
-end
\ No newline at end of file
+end
diff --git a/lib/helpers.rb b/lib/helpers.rb
index 5638417..09cb2cf 100644
--- a/lib/helpers.rb
+++ b/lib/helpers.rb
@@ -20,8 +20,8 @@ def options
     {
       "Cache-Control" => "max-age=315360000, public",
       "Expires" => "Sun, 29 Jun 2036 17:48:34 GMT",
-      "x-amz-storage-class" => "REDUCED_REDUNDANCY"
+      "x-amz-storage-class" => ENV["AWS_S3_STORAGE_CLASS"] || "REDUCED_REDUNDANCY",
     }
   end
 
-end
\ No newline at end of file
+end
diff --git a/lib/s3_pool.rb b/lib/s3_pool.rb
index ddfdcee..8fbc741 100644
--- a/lib/s3_pool.rb
+++ b/lib/s3_pool.rb
@@ -3,6 +3,10 @@
     provider: "AWS",
     aws_access_key_id: ENV["AWS_ACCESS_KEY_ID"],
     aws_secret_access_key: ENV["AWS_SECRET_ACCESS_KEY"],
-    persistent: true
+    region: ENV["AWS_S3_REGION"] || "us-east-1",
+    host: ENV["AWS_S3_HOST"] || "s3.amazonaws.com",
+    endpoint: ENV["AWS_S3_ENDPOINT"] || nil,
+    path_style: ENV["AWS_S3_PATH_STYLE"] || false,
+    persistent: true,
   )
 end
